lib/wanda/catalog.ex

```diff
@@ -53,14 +53,29 @@ defmodule Wanda.Catalog do
   @doc """
   Get specific checks from the catalog.
   """
-  @spec get_checks([String.t()]) :: [Check.t()]
-  def get_checks(checks_id) do
-    Enum.flat_map(checks_id, fn check_id ->
+  @spec get_checks([String.t()], map()) :: [Check.t()]
+  def get_checks(checks_id, env) do
+    checks_id
+    |> Enum.flat_map(fn check_id ->
       case get_check(check_id) do
         {:ok, check} -> [check]
         {:error, _} -> []
       end
     end)
+    |> Enum.filter(&when_condition(&1, env))
+  end
+
+  defp when_condition(_, env) when env == %{}, do: true
+
+  defp when_condition(%Check{when: nil}, _), do: true
+
+  defp when_condition(%Check{when: when_clause}, env) do
+    scope = %{"env" => env}
```

Comment:
Tiny nitpick; why do you place the map into a temporary variable `scope` here? Why not just `case Rhai.eval(when_clause, %{"env" => env}) do` as it's not used anywhere else?

Reply:
You're right, didn't seem like a good one to me, then your comment made me change my mind :smile: